### PR TITLE
[Merged by Bors] - chore: rename Name.getString -> Name.lastComponentAsString

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

File: Mathlib/Lean/Expr/Basic.lean [hunk 1]
```diff
@@ -64,13 +64,16 @@ def updateLast (f : String → String) : Name → Name
   | .str n s => .str n (f s)
   | n        => n
 
+
```
Comment (dwrensha) [inline Mathlib/Lean/Expr/Basic.lean:67]: Was this new empty line intentional?

**Review by dwrensha — APPROVED** [review]
LGTM!

**Comment by dwrensha** [comment]
Perhaps the current name was meant as a parallel to the existing `Lean.Name.getString!` in core? https://github.com/leanprover/lean4/blob/61a84c96db4ef76ebcd7b1510c5b7c7f1cf81212/src/Lean/Data/Name.lean#L20-L22
(In any case, I think your change in this PR makes it a lot clearer.)

File: Mathlib/Lean/Expr/Basic.lean (hunk 1 above)
Comment (semorrison) [inline Mathlib/Lean/Expr/Basic.lean:67]:
```suggestion
```

Comment (semorrison) [reply Mathlib/Lean/Expr/Basic.lean]: Oops, too much copy and paste.

**Comment by semorrison** [comment]
> Perhaps the current name was meant as a parallel to the existing `Lean.Name.getString!` in core? [leanprover/lean4@`61a84c9`/src/Lean/Data/Name.lean#L20-L22](https://github.com/leanprover/lean4/blob/61a84c96db4ef76ebcd7b1510c5b7c7f1cf81212/src/Lean/Data/Name.lean#L20-L22) (In any case, I think your change in this PR makes it a lot clearer.)

Ah, good find! I'll see if I can rename that, too.

**Review by jcommelin — COMMENTED** [review]
I slightly prefer the shorter `lastAsString` suggestion that you made on zulip.

Whatever you choose, this is an improvement.

bors d+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**